app/models/link.rb

```diff
@@ -0,0 +1,45 @@
+class Link < ActiveRecord::Base
+  belongs_to :user
+
+  validates_presence_of :actual
+  validates :vanity_string, uniqueness: true
+  after_create :generate_vanity_string
+
+  ALPHABETS = "x6QW9Js5T7MHCyVj0uRIkrYaLFlSh4b3fpZANz1o8wqKtX2d"\
```

Comment:
`sample` method on ALPHABETS will increase drastically when the database id increases. Thus the use of encoding database ID to base_64.